projects/observability/src/shared/components/cartesian/chart-builder.service.ts

```diff
@@ -12,7 +13,22 @@ export class ChartBuilderService {
     protected readonly d3Utils: D3UtilService
   ) {}
 
-  public build<TData>(strategy: RenderingStrategy, element: Element, renderer: Renderer2): CartesianChart<TData> {
-    return new DefaultCartesianChart(element, this.injector, strategy, this.svgUtilService, this.d3Utils, renderer);
+  public build<TData>(
+    strategy: RenderingStrategy,
+    element: Element,
+    renderer: Renderer2,
+    groupId?: string,
+    chartSyncService?: ChartSyncService<TData>
```

Comment:
This need not be passed in, the whole injector is already in the chart so it can look it up if needed without the extra arg.